### PR TITLE
Use uv in CI to speed-up requirements install

### PR DESCRIPTION
**Pull request by Wauplin**
Quick benchmarks show it saves around 30s-40s per workflow (both for code quality and CI tests)

**Note:** running `uv` instead of `pip` in Github CI is not yet a first-class citizen but should be the case once https://github.com/astral-sh/uv/issues/1526 is addressed. This is why we need to source env in each step.

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Awesome, love `uv` !

File: .github/workflows/python-quality.yml [hunk 1]
```diff
@@ -23,20 +23,26 @@ jobs:
...
-      - run: ruff format --check tests src contrib # formatter
-      - run: python utils/check_contrib_list.py
-      - run: python utils/check_static_imports.py
-      - run: python utils/generate_async_inference_client.py
-      - run: python utils/generate_inference_types.py
+          pip install --upgrade uv
+          uv venv
+
+      - name: Install dependencies
+        run: uv pip install "huggingface_hub[dev] @ ."
```
Comment (charliermarsh) [inline .github/workflows/python-quality.yml:35]: You _should_ be able to do `uv pip install --system "huggingface_hub[dev] @ ."` on the latest release, if you want to remove the virtualenv (per the PR summary), but using a virtualenv is also good.

Comment (Wauplin) [reply .github/workflows/python-quality.yml:35]: Thanks @charliermarsh! I just saw the issue has been merged on uv's repo :tada:. Thanks for the direct ping here :heart:

Comment (charliermarsh) [reply .github/workflows/python-quality.yml:35]: Of course! It's pretty new so of course ping if you see issues with it.

**Review by ArthurZucker — COMMENTED** [review]
Nice! We'll also do something similar in transformers! Speeds up a lot the resolution as well Related PR: https://github.com/huggingface/transformers/pull/29202

**Pull request merged**